src/components/Toggle/Toggle.tsx

```diff
@@ -75,8 +76,12 @@ const ToggleButton = ({ className, checked, ...other }: ToggleButtonProps) => (
 /**
  * In order to use the individual <Toggle.Label> and <Toggle.Button> subcomponents for customization,
  * they need to be wrapped in this <ToggleWrapper> component for some functionality HeadlessUI provides.
+ *
+ * Instead of direct assignment to ToggleWrapper, wrapping into a custom component prevents TypeScript erroring of using private Headless Switch Group.
  */
-const ToggleWrapper = Switch.Group;
+const ToggleWrapper = (props: ExtractProps<typeof Switch.Group>) => (
+  <Switch.Group {...props} />
+);
```

Comment:
As stated in the comments, this wrapping prevents type erroring that occurs with direct assignment and re-export, similar as https://github.com/chanzuckerberg/edu-design-system/blob/main/src/components/Popover/Popover.tsx#L106-L111

Reply:
Weird